activerecord/lib/active_record/connection_adapters/postgresql/schema_definitions.rb

```diff
@@ -45,130 +45,780 @@ def primary_key(name, type = :primary_key, **options)
 
         ##
         # :method: citext
         # :call-seq: citext(*names, **options)
+        #
+        # Adds a +citext+ column for storing case-insensitive text values.
+        #
+        # This is backed by PostgreSQL’s +citext+ extension, which treats values
+        # as case-insensitive for comparisons and indexing.
+        #
+        # Example:
+        #
+        #   t.citext :email
+        #
+        # This creates a column named +email+ that will treat values like "User@example.com"
+        # and "user@example.com" as equal.
 
         ##
         # :method: daterange
         # :call-seq: daterange(*names, **options)
+        #
+        # Adds a +daterange+ column which allows storage and querying
+        # of inclusive or exclusive date intervals.
+        #
+        # This type is mapped to {Ruby Range}[https://docs.ruby-lang.org/en/master/Range.html] objects.
+        #
+        # Example:
+        #   t.daterange :duration
+        #
+        # This creates a column named +duration+ that can store ranges like:
+        #   '2024-01-01'..'2024-12-31'
+        #
+        #   irb> Event.create(duration: Date.new(2014, 2, 11)..Date.new(2014, 2, 12))
+        #
+        #   irb> event = Event.first
+        #   irb> event.duration
+        #   # => Tue, 11 Feb 2014...Thu, 13 Feb 2014
+        #
+        # For all events on a given date:
+        #
+        #   irb> Event.where("duration @> ?::date", Date.new(2014, 2, 12))
+        #
+        # Working with range bounds:
+        #
+        #   irb> event = Event.select("lower(duration) AS starts_at").select("upper(duration) AS ends_at").first
+        #
+        #   irb> event.starts_at
+        #   # => Tue, 11 Feb 2014
+        #
+        #   irb> event.ends_at
+        #   # => Thu, 13 Feb 2014
+        #
+        # 🔗 See also: {PostgreSQL type definition}[https://www.postgresql.org/docs/current/static/rangetypes.html] and PostgreSQL range functions and operators}[https://www.postgresql.org/docs/current/static/functions-range.html]
+
+        ##
+        # :method: enum
+        # :call-seq: enum(*names, **options)
+        #
+        # Adds an +enum+ column for storing enumerated values.
+        #
+        # The PostgreSQL +enum+ type can be used directly as a column, or mapped to
+        # an {ActiveRecord::Enum}[https://api.rubyonrails.org/classes/ActiveRecord/Enum.html]
+        # to add model-level helpers and validations.
+        #
+        # Example:
+        #
+        # Creating an enum type and using it in a new table:
+        #
+        #   # db/migrate/20131220144913_create_articles.rb
+        #   def change
+        #     create_enum :article_status, ["draft", "published", "archived"]
+        #
+        #     create_table :articles do |t|
+        #       t.enum :status, enum_type: :article_status, default: "draft", null: false
+        #     end
+        #   end
+        #
+        # You can also create the enum and add it to an existing table:
```

Comment:
Avoiding "you", maybe something like:
```suggestion
        # The enum can also be added it to an existing table:
```